script.js

```diff
@@ -0,0 +1,68 @@
+// 1 Написать функцию, преобразующую число в объект. Передавая на вход число от 0 до 999, надо получить на выходе объект, в котором в соответствующих свойствах описаны единицы, десятки и сотни. Например, для числа 245 надо получить следующий объект: {‘единицы’: 5, ‘десятки’: 4, ‘сотни’: 2}. Если число превышает 999, необходимо выдать соответствующее сообщение с помощью console.log и вернуть пустой объект
+
+
+
+var number = +prompt('Введи 3х значное число');
+var numberPayment =  [{
+    numberHungret:0,    
+    numberTen:0,
+    numberUnit:0,    
+};
+
+function scatter(number){    
+
+    if ( number > 999 || number < 99 || isNaN(number) ) { 
+
+        console.log('Число не подходит');
+        number = +prompt('Введи 3х значное число');                
+        scatter(number);     
+        
+    }else {
+
+        numberPayment.numberUnit = number % 10;
+        numberPayment.numberTen = (number % 100 - numberPayment.numberUnit) / 10;
+        numberPayment.numberHungret = (number %1000-(number % 100))/100 ;
+
+        console.log(numberPayment);
+    }
+
+    return numberPayment;    
+
+}
+
+scatter(number);
+
+
+
+//  a) В прошлом домашнем задании вы реализовали корзину на базе массивов. Какими объектами можно заменить их элементы?
+
+var cart  = [
+        {towarName:'Котеров, Симдянов: PHP 7', cost:1080},
+        {towarName:'ПК для чайников. Краткий справочник',cost: 241},
+        {towarName:'Дьявольские куклы мадам Мэндилип',cost: 388},
+        {towarName:'Война и мир',cost: 783}
+    ];
+    
+    
+    // b) Перенести функционал подсчета корзины на объектно-ориентированную базу.
+    
+function countBasketPrice(cart) {
+
+    var cartPrice = 0;
+
+    for(var i = 0 in cart) {
+        if(isNaN(cart[i].cost)) continue; 
```

Comment:
Лучше всегда ставить скобки - так лучше читается
```suggestion
        if(isNaN(cart[i].cost)) {
          continue; 
        }
```